core/src/main/java/org/polypheny/db/type/entity/PolyInterval.java

```diff
@@ -27,32 +28,95 @@
 import org.jetbrains.annotations.Nullable;
 import org.polypheny.db.nodes.IntervalQualifier;
 import org.polypheny.db.type.PolySerializable;
+import org.polypheny.db.type.PolyType;
+import org.polypheny.db.type.entity.category.PolyNumber;
+import org.polypheny.db.util.temporal.TimeUnit;
 
 @EqualsAndHashCode(callSuper = true)
 @Value
 @Slf4j
+@NonFinal
 public class PolyInterval extends PolyValue {
 
 
-    public BigDecimal value;
-    public IntervalQualifier qualifier;
+    @NotNull
+    public Long millis;
+
+    @Getter
+    @NotNull
+    public Long months;
 
 
     /**
-     * Creates a PolyInterval.
+     * Creates a PolyInterval, which includes a millis and a month value, to allow for combinations liki 7-1 Month to Day, which is 7 months and 7 days.
+     * And required for example by the SQL standard.
      *
-     * @param value The amount of the range
-     * @param qualifier The unit qualifier, e.g. YEAR, MONTH, DAY, etc.
+     * @param millis millis since Epoch.
+     * @param months months since Epoch.
      */
-    public PolyInterval( BigDecimal value, IntervalQualifier qualifier ) {
-        super( qualifier.typeName() );
-        this.value = value;
-        this.qualifier = qualifier;
+    public PolyInterval( @NotNull Long millis, @NotNull Long months ) {
+        super( PolyType.INTERVAL );
+        this.millis = millis;
+        this.months = months;
+    }
+
+
+    private static MonthsMilliseconds normalize( Long value, TimeUnit unit ) {
+        if ( unit == TimeUnit.YEAR ) {
+            return new MonthsMilliseconds( value * 12, 0 );
+        } else if ( unit == TimeUnit.MONTH ) {
+            return new MonthsMilliseconds( value, 0 );
+        } else if ( unit == TimeUnit.DAY ) {
+            return new MonthsMilliseconds( 0, value * 24 * 60 * 60 * 1000 );
+        } else if ( unit == TimeUnit.HOUR ) {
+            return new MonthsMilliseconds( 0, value * 60 * 60 * 1000 );
+        } else if ( unit == TimeUnit.MINUTE ) {
+            return new MonthsMilliseconds( 0, value * 60 * 1000 );
+        } else if ( unit == TimeUnit.SECOND ) {
+            return new MonthsMilliseconds( 0, value * 1000 );
+        } else if ( unit == TimeUnit.MILLISECOND ) {
+            return new MonthsMilliseconds( 0, value );
+        } else {
+            throw new NotImplementedException( "since Epoch" );
```

Comment:
??

Reply:
Replaced with adjusted exception and error message.